corehq/apps/cloudcare/utils.py

```diff
@@ -6,10 +6,70 @@
 from six.moves.urllib.parse import quote
 
 from corehq import toggles
-from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.apps.app_manager.dbaccessors import (
+    get_app_ids_in_domain,
+    get_apps_in_domain,
+    get_latest_build_doc,
+    get_latest_build_id,
+    get_latest_released_app_doc,
+    get_latest_released_build_id,
+)
 from corehq.util.quickcache import quickcache
 
 
+def can_user_access_web_app(app, user, domain):
+    """
+    :param app: app doc (not wrapped Application)
+    :param user: either a WebUser or CommCareUser
+    :param domain: domain name
+    """
+    # Backwards-compatibility - mobile users haven't historically required this permission
+    has_access_via_permission = False
+    if user.is_web_user() or user.can_access_any_web_apps(domain):
+        has_access_via_permission = user.can_access_web_app(domain, app.get('copy_of', app.get('_id')))
```

Comment:
See 0ddf5addceae5d9cd68b87172db60d33c405e745

Reply:
I resolved this prematurely. I don't have a good answer to the possibility of `app_id` being None. I'd imagine it makes sense to change `app.get('_id')` to `app['_id']` so that it fails hard.